cr_module/classes/redfish.py

```diff
@@ -226,7 +227,51 @@ def restore_session_from_file(self):
         self.session_was_restored = True
 
         return
-
+    
+    def remove_session_lock(self):
+        if (self.cli_args.sessionlock is False or self.session_file_path is None):
+            return
+        try:
+            if os.path.exists(self.session_file_lock):
+                os.remove(self.session_file_lock)
+        except Exception as e:
+            self.exit_on_error(f"Unable to remove session lock '{self.session_file_lock}': {e}", "CRITICAL")
+        return
+    
+    def is_session_locked(self):
+        if (self.cli_args.sessionlock is False or self.session_file_path is None):
+            return False
+        if os.path.exists(self.session_file_lock) is False:
+            return False
+        return os.path.exists(self.session_file_lock)
+        pid = None
+        try:
+            with open(self.session_file_lock, 'r') as lock_file:
+                pid = lock_file.read().strip()
+        except Exception as e:
+            self.exit_on_error(f"Unable to read session lock '{self.session_file_lock}': {e}", "CRITICAL")
+        if isinstance(pid, int) is None:
+            self.exit_on_error(f"Session lock is empty '{self.session_file_lock}'", "CRITICAL")
+        try:
+            # Sending signal 0 does not kill the process
+            # It is used to check if a process with the given PID exists
+            print(pid)
+            os.kill(int(pid), 0)
+        except OSError:
+            return False  # The process is not running
+        return True # The process is running
+    
+    def write_session_lock(self):
+        if (self.cli_args.sessionlock is False or self.session_file_path is None):
+            return
+        if os.path.exists(self.session_file_lock) is False:
+            try:
+                with open(self.session_file_lock, 'w') as file:
+                    file.write(str(os.getpid()))
+            except Exception as e:
+                self.exit_on_error(f"Unable to write session lock '{self.session_file_lock}': {e}", "CRITICAL")
```

Comment:
Plugin internal errors should return UNKNOWN

Reply:
True, i corrected it